client/blocks/upload-image/index.jsx

```diff
@@ -0,0 +1,179 @@
+import DropZone from 'components/drop-zone';
+
+class UploadImage extends Component {
+	state = {
+		isEditingImage: false,
+		image: false,
+	};
+
+	static propTypes = {
+		isUploading: PropTypes.bool,
+		imageEditorModalClassNames: PropTypes.string,
+		allowedAspectRatios: PropTypes.arrayOf( PropTypes.oneOf( AspectRatiosValues ) ),
+		texts: PropTypes.object,
+		onImageEditorDone: PropTypes.func,
+	};
+
+	static defaultProps = {
+		allowedAspectRatios: [ AspectRatios.ASPECT_1X1 ],
+		texts: {},
+		backgroundContent: null,
+	};
+
+	onReceiveFile = ( files ) => {
+		const extension = path.extname( files[ 0 ].name )
+			.toLowerCase()
+			.substring( 1 );
+
+		if ( ALLOWED_FILE_EXTENSIONS.indexOf( extension ) === -1 ) {
+			return;
+		}
+
+		const imageObjectUrl = URL.createObjectURL( files[ 0 ] );
+
+		this.setState( {
+			isEditingImage: true,
+			image: imageObjectUrl
+		} );
+	};
+
+	onImageEditorDone = ( error, imageBlob ) => {
+		this.hideImageEditor();
+
+		if ( error ) {
+			return;
+		}
+
+		this.props.onImageEditorDone( imageBlob );
+	};
+
+	hideImageEditor = () => {
+		const {
+			resetAllImageEditorState: resetAllImageEditorStateAction
+		} = this.props;
+
+		resetAllImageEditorStateAction();
+
+		URL.revokeObjectURL( this.state.image );
+
+		this.setState( {
+			isEditingImage: false,
+			image: false
+		} );
+	};
+
+	renderImageEditor() {
+		const {
+			imageEditorModalClassNames,
+			allowedAspectRatios,
+			texts,
+		} = this.props;
+
+		const classes = classnames( 'upload-image-modal', imageEditorModalClassNames );
+
+		if ( this.state.isEditingImage ) {
+			return (
+				<Dialog
+					additionalClassNames={ classes }
+					isVisible={ true }
+				>
+					<ImageEditor
+						allowedAspectRatios={ allowedAspectRatios }
+						media={ { src: this.state.image } }
+						onDone={ this.onImageEditorDone }
+						onCancel={ this.hideImageEditor }
+						doneButtonText={ texts.doneButtonText ? texts.doneButtonText : 'Done' }
+					/>
+				</Dialog>
+			);
+		}
+	}
+
+	render() {
+		const {
+			backgroundContent,
+			isUploading,
+			translate,
+			texts,
+		} = this.props;
+
+		let {
+			placeholderContent,
+			uploadingContent,
+		} = this.props;
+
+		if ( ! placeholderContent ) {
+			placeholderContent = (
+				<div className="upload-image__placeholder">
+						<Gridicon icon="add-image" size={ 36 } />
+					<span>Add an image</span>
+				</div>
+			);
+		}
+
+		if ( ! uploadingContent ) {
+			uploadingContent = ( <Spinner className="upload-image__spinner" /> );
+		}
+
+		return (
+			<div className={	classnames( 'upload-image' ) } >
+				{ this.renderImageEditor() }
+
+				<FilePicker accept="image/*" onPick={ this.onReceiveFile }>
+					<div
+						className={
+							classnames( 'upload-image__image-container',
+								{ 'is-uploading': isUploading }
+							)
+						}
+					>
+						<DropZone
+							textLabel={ texts.dragUploadText
+								? texts.dragUploadText
+								: translate( 'Drop to upload image' )
```

Comment:
Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Drop images to upload' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).